### DISCUSSION TITLE
Add Adjustable Margin Support

### DISCUSSION THREAD

**Discussion by a-dows** [rxhanson/Rectangle#848] [Ideas]
I was a long-time user of Divvy, by Mizage, for windows management and it had a feature I have not found in any alternative—adjustable margins.

Adding margins helps distinguish between windows and allows the desktop wallpaper to be visible around the snapped windows.

<img width="263" alt="Screen Shot 2022-07-06 at 8 41 31 AM" src="https://user-images.githubusercontent.com/42701400/177552751-36355e0b-ccc6-407f-a40f-83badb7c0d39.png">

**Comment by Aaron-Rumpler** [comment]
Rectangle has a 'Gaps between windows' option that does this:

<img width="602" alt="Rectangle - Gaps between windows" src="https://user-images.githubusercontent.com/80610725/177635530-9cc70677-c1f6-4d22-93d6-a74713f55043.png">

<img width="1680" alt="Screen Shot 2022-07-07 at 8 23 16 AM" src="https://user-images.githubusercontent.com/80610725/177636475-9776fb36-01df-4522-8bf6-5cb8df1d24ed.png">

There's also a couple of [Terminal Commands](https://github.com/rxhanson/Rectangle/blob/master/TerminalCommands.md) for further customising the behaviour:

- [Setting gaps at the screen edges](https://github.com/rxhanson/Rectangle/blob/master/TerminalCommands.md#setting-gaps-at-the-screen-edges)
- [Disabling gaps when maximizing](https://github.com/rxhanson/Rectangle/blob/master/TerminalCommands.md#disabling-gaps-when-maximizing)

**Reply by wolterkam** [reply]
Setting the screen edge gaps really should be available as an option in the graphical preferences.

This is something that's worth looking into especially now considering that Stage Manager appears to be built around the idea of clustering the windows in the _center of the screen_, while Rectangle by default implements a grid that distributes the windows evenly on the _entire screen_.

Allowing for "scaling down" the space where Rectangle's grid layout is drawn could be implemented with a single slider which affects all _screenEdgeGap*_ variables equally.

**Reply by rxhanson** [reply]
A common use for screen edge gaps is to accommodate 3rd party docks, which would make a single slider not ideal. I have other items prioritized above this, but I would actually welcome a good PR for adding in the number inputs like the Divvy screenshot.

**Comment by a-dows** [comment]
Thank you! Sorry for missing this.
